src/DiscoveryClient.coffee

```diff
@@ -48,29 +71,68 @@ class DiscoveryClient
         else
           Promise.reject(e)
 
+  connect:(callback) =>
+    @_connect(callback, @discoveryConnector, @serverList, @regionsConfig.discoverRegion)
+
+    for region, connectors of @regions
+      @_connect(null, connectors.discoveryConnector, connectors.serverList, region)
+
+  getUrlFromRegion:(baseUrl, regionName ) ->
+    baseUrl.replace '*', regionName
+ 
+  getHostUrl: () =>
+    @getUrlFromRegion @host, @regionsConfig.discoverRegion
+
   reconnect:(callback)=>
     @connect(callback)
 
   stopAnnouncementHeartbeat:()=>
-    if @heartbeatIntervalId
-      clearInterval(@heartbeatIntervalId)
+    if @heartbeatIntervalIds
+      for intervals in @heartbeatIntervalIds
+        clearInterval(intervals)
+      @heartbeatIntervalIds = []
 
   startAnnouncementHeartbeat:()=>
     @stopAnnouncementHeartbeat()
+    if not @heartbeatIntervalIds
+      @heartbeatIntervalIds = []
 
-    @heartbeatIntervalId = setInterval(
+    @heartbeatIntervalIds.push setInterval(
       @discoveryAnnouncer.pingAllAnnouncements
       @HEARTBEAT_INTERVAL_MS
     )
+
+    for region,connectors of @regions
+      @heartbeatIntervalIds.push setInterval(
+        connectors.discoveryAnnouncer.pingAllAnnouncements
+        @HEARTBEAT_INTERVAL_MS
+      )
+
     return
 
-  announce:(announcement, callback)->
-    @discoveryAnnouncer.announce(
-      announcement, callback)
+  announce:(announcement, callback)=>
+    announcement.environment = @regionsConfig.discoverRegion
+
+    announcementCalls = []
+
+    announcementCalls.push (nodeCB) =>
+      @discoveryAnnouncer.announce announcement,nodeCB
+
+    for region, connectors of @regions
+      announcementCalls.push (_asyncCB) ->
+        connectors.discoveryAnnouncer.announce announcement,_asyncCB
+
+
+    async.parallel announcementCalls, (items) ->
+      if callback
+        callback()
```

Comment:
we should use promises here rather than callbacks, bluebird has ability to run promises in parallel, and the error handling will be easier to code